mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1585,6 +1727,32 @@ vectorizeLinalgOpPrecondition(LinalgOp linalgOp,
   return success();
 }
 
+/// TODO: Use a matcher to check for a constant padding value.
+static LogicalResult
+vectorizePackOpPrecondition(tensor::PackOp packOp,
+                            ArrayRef<int64_t> inputVectorSizes) {
+  auto padValue = packOp.getPaddingValue();
+  if (padValue && !padValue.getDefiningOp<arith::ConstantOp>()) {
+    LDBG("pad value is not constant: " << packOp << "\n");
+    return failure();
+  }
+
```

Comment:
I realized there is no matcher for constant float values like there is for constant int values. I didn't know exactly how this should be done.
https://github.com/llvm/llvm-project/blob/07bf1ddb4eb0abfff20542fd4459bace1f72107f/mlir/lib/Dialect/Utils/StaticValueUtils.cpp#L108-L122
For int, getConstantIntValue passes an uninitialized APSInt. For the analogous float matcher, I'm not sure the best way to implement a similar function, since there are several possible floating point semantics. I suppose there could be a util function that simply checks if the value is constant and does not return the value. I think then an arbitrary float semantic could be used.

Reply:
good catch, I think the current implementation is okay, which checks if they are from arith.constant.

Reply:
Hey Max, I just found this might address the concern.

```cpp
// Return the constant attribute, or null if the Operation isn't a constant.
Attribute getConstantAttr(Operation *constantOp) {
  Attribute constant;
  matchPattern(value.getDefiningOp(), m_Constant());
  return constant;
}
```

https://mlir.llvm.org/getting_started/Faq/#many-dialects-define-a-constant-operation-how-do-i-get-a-constant-value-generically